internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -736,3 +782,72 @@ func getComposerResourcesNames(cds []ComposedResource) []string {
 	}
 	return names
 }
+
+type compositionResultMeta struct {
+	numWarningEvents   int
+	conditionTypesSeen map[xpv1.ConditionType]struct{}
+}
+
+func (r *Reconciler) handleCommonCompositionResult(ctx context.Context, res CompositionResult, xr *composite.Unstructured, log logging.Logger) compositionResultMeta {
```

Comment:
Do you need `log` as an argument here? I think there's an `r.log` field of the `Reconciler` struct.

Reply:
Ah right.

Given we pass `xr` to this method my preference would be to use `r.log` and repeat this `WithValues` at the top of the method, rather than passing the logger as an argument.

Reply:
We would lose out on the `request` part of the logger, unless if we pass `req` to the function. Thoughts?

Reply:
I don't love it but I'd probably err toward losing that rather than passing the logger as an argument.